### PR TITLE
Slug attempt no gem

### PR DESCRIPTION
**Pull request by cmacaulay**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,5 +50,9 @@ group :development do
   gem 'listen', '~> 3.0.5'
 end
 
+group :test do
```
Comment (cmacaulay) [inline Gemfile:53]: Configures database cleaner.

File: app/views/categories/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<h1><%= @category.name %></h1>
```
Comment (cmacaulay) [inline app/views/categories/show.html.erb:1]: Adds show view - very similar to items index, we should probably make a partial.

File: db/migrate/20170222185641_add_slug_to_categories.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddSlugToCategories < ActiveRecord::Migration[5.0]
+  def change
+    add_column :categories, :slug, :string
+    add_index :categories, :slug
+  end
+end
```
Comment (cmacaulay) [inline db/migrate/20170222185641_add_slug_to_categories.rb:6]: this adds a column to categories that allows us to have a slug.

File: db/schema.rb [hunk 4]
```diff
@@ -10,13 +10,15 @@
...
-ActiveRecord::Schema.define(version: 20170222010254) do
+ActiveRecord::Schema.define(version: 20170222185641) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
   create_table "categories", force: :cascade do |t|
     t.string "name"
+    t.string "slug"
+    t.index ["slug"], name: "index_categories_on_slug", using: :btree
```
Comment (cmacaulay) [inline db/schema.rb:21]: The index will make it go faster!

File: app/models/category.rb [hunk 5]
```diff
@@ -1,4 +1,11 @@
 class Category < ApplicationRecord
+
   validates :name, presence: true, uniqueness: true
-  has_many :items 
+  has_many :items
+  before_save :assign_slug
```
Comment (cmacaulay) [inline app/models/category.rb:5]: this method allows us to save the path "category-name" as the slug.

**Review by cmacaulay — COMMENTED** [review]
@CjMoore @meyerhoferc @valerietrudell ready for review!

**Comment by meyerhoferc** [comment]
Does this pass tests?
Also, can we fix the hound complaints that aren't about long lines?

File: app/views/shared/_nav.html.erb [hunk 6]
```diff
@@ -1,6 +1,7 @@
 <nav>
    <div class="nav-wrapper orange darken-3">
-     <a href="#" class="brand-logo">Weasleys' Wizard Wheezes</a>
+     <a href="<%= root_path %>" class="brand-logo">Weasleys' Wizard Wheezes</a>
+     <a href="/cart" class="brand-logo">View Cart</a>
```
Comment (CjMoore) [inline app/views/shared/_nav.html.erb]: I'm going to take out this line, I think it'll add it back in when we merge.

**Pull request merged**